crates/ra_ide/src/completion/complete_unqualified_path.rs

```diff
@@ -4,17 +4,42 @@ use hir::ScopeDef;
 use test_utils::tested_by;
 
 use crate::completion::{CompletionContext, Completions};
-use ra_syntax::AstNode;
+use hir::Adt;
+use ra_syntax::SyntaxNode;
+use ra_syntax::{ast, AstNode, SyntaxKind};
 
 pub(super) fn complete_unqualified_path(acc: &mut Completions, ctx: &CompletionContext) {
-    if !ctx.is_trivial_path {
```

Comment:
not sure about removing this condition -- if we already have a qualified path like `Enum::Qu<|>`, there's other completions that will complete the enum variant

Reply:
Oh, and we don't need to worry about `use Enum::*`; in that case the normal unqualified path completion will already do the job :)

Reply:
Regarding the `is_trivial_path` condition, I found that whenever the cursor is in a pattern position the `is_trivial_path` is set to `false`. I think this is because [here](https://github.com/rust-analyzer/rust-analyzer/blob/7ab28cacbb50d78e4e6f12293351683625cab607/crates/ra_ide/src/completion/completion_context.rs#L296-L297) the `CompletionContext` only checks the path and sets that flag if the current node is a `NameRef`, and it doesn't do so for `BindPat`s.

For this PR I changed the condition to `return` early only if it is not a trivial path _and_ not a pattern.

Reply:
> I found that whenever the cursor is in a pattern position the is_trivial_path is set to false.

Hmm, that might be a bug -- I vaguely recall frequently being annoyed about not getting completions I'd expect in patterns, though I haven't looked into it :thinking: